src/main/scala/mg7/package.scala

```diff
@@ -23,44 +21,23 @@ package object mg7 {
   def parseLong(str: String): Option[Long] = util.Try(str.toLong).toOption
   def parseDouble(str: String): Option[Double] = util.Try(str.toDouble).toOption
 
+  // TODO why this?
   def lookup[A, B](a: A, m: Map[A, B]): (A, B) = a -> m.apply(a)
 
-  def maximums[T, X](s: Iterable[T])(f: T => X)
-    (implicit cmp: Ordering[X]): List[T] =
-      s.foldLeft(List[T]()) {
-        case (a :: acc, t) if (    cmp.lt(f(t), f(a)) ) => a :: acc
-        case (a :: acc, t) if ( cmp.equiv(f(t), f(a)) ) => t :: a :: acc
-        // either acc is empty or t is the new maximum
-        case (_, t) => List(t)
-      }
-
+  // TODO why not create an ordering from `f` and use std library?
+  def maximums[T, X](s: Iterable[T])(f: T => X)(implicit cmp: Ordering[X]): List[T] =
+    s.foldLeft(List[T]()) {
+      case (a :: acc, t) if (    cmp.lt(f(t), f(a)) ) => a :: acc
+      case (a :: acc, t) if ( cmp.equiv(f(t), f(a)) ) => t :: a :: acc
+      // either acc is empty or t is the new maximum
+      case (_, t) => List(t)
+    }
+  // TODO why not as ops?
```

Comment:
better to write implicit classes for this. This method has no relation with the mg7 package; if we need to extend Iterable/Traversable/whatever, better do it properly.